pkg/cluster/base.go

```diff
@@ -74,14 +80,15 @@ const (
 
 // CreateClusterRequest describes a create cluster request
 type CreateClusterRequest struct {
-	Name       string                   `json:"name" yaml:"name" binding:"required"`
-	Location   string                   `json:"location" yaml:"location"`
-	Cloud      string                   `json:"cloud" yaml:"cloud" binding:"required"`
-	SecretId   string                   `json:"secretId" yaml:"secretId"`
-	SecretIds  []string                 `json:"secretIds,omitempty" yaml:"secretIds,omitempty"`
-	SecretName string                   `json:"secretName" yaml:"secretName"`
-	PostHooks  PostHooks                `json:"postHooks" yaml:"postHooks"`
-	Properties *CreateClusterProperties `json:"properties" yaml:"properties" binding:"required"`
+	Name                 string                   `json:"name" yaml:"name" binding:"required"`
+	Location             string                   `json:"location" yaml:"location"`
+	Cloud                string                   `json:"cloud" yaml:"cloud" binding:"required"`
+	SecretId             string                   `json:"secretId" yaml:"secretId"`
+	SecretIds            []string                 `json:"secretIds,omitempty" yaml:"secretIds,omitempty"`
+	SecretName           string                   `json:"secretName" yaml:"secretName"`
+	PostHooks            PostHooks                `json:"postHooks" yaml:"postHooks"`
+	Properties           *CreateClusterProperties `json:"properties" yaml:"properties" binding:"required"`
+	ProductDetailsGetter cloudinfo.ProductDetailsGetter
```

Comment:
**Note**: This feels like being in the wrong place here.
It should not be part of the request, because the request is supposed to be a descriptive type of the sent in request.
Let me think a bit about alternatives.

TODO: @pregnor

Reply:
I agree this is not the perfect place, the easy alternative would be to pass down the `productDetailsGetter` as function parameter all the way where we would use it...